packages/editor/src/pages/Editor/components/Main/Editor/ReactMonaco.tsx

```diff
@@ -92,9 +94,14 @@ export class ReactMonaco extends Component<IProps, IState> {
   private getUri = () =>
     monaco.Uri.file(`${this.props.solutionId}/${this.props.file.id}`);
 
-  private getModel = () => {
+  private getModel = (options = { recreateFromScratch: false }) => {
```

Comment:
I think the cleaner way to express this would be via the URI that monaco is looking at. Incorporate the language into the uri, and monaco will already "recreateFromScratch"